runners/apex/src/main/java/org/apache/beam/runners/apex/translators/functions/ApexGroupByKeyOperator.java

```diff
@@ -230,7 +232,7 @@ private void processElement(WindowedValue<KV<K, V>> windowedValue) throws Except
     }
     StateInternals<K> stateInternals = perKeyStateInternals.get(keyBytes);
     if (stateInternals == null) {
-      stateInternals = InMemoryStateInternals.forKey(key);
+      stateInternals = stateInternalsFactory.stateInternalsForKey(key);
```

Comment:
It cannot be transient because when the operator fails and get redeployed in a different process we still want it to remember its state.